axlearn/cloud/gcp/job.py

```diff
@@ -639,6 +639,327 @@ def _execute(self) -> Any:
+        cfg.accelerator.set(instance_type=fv.instance_type, num_replicas=fv.num_replicas)
+        return cfg
+
+    def __init__(self, cfg: Config):
+        bundler_cfg = cfg.bundler
+        bundler_cfg = getattr(bundler_cfg, "inner", bundler_cfg)
+        if bundler_cfg is None or not issubclass(bundler_cfg.klass, BaseDockerBundler):
+            raise NotImplementedError(f"Only docker bundler supported, got: {bundler_cfg}")
+        super().__init__(cfg)
+
+    def _build_sidecar_container(self) -> Nested[Any]:
+        cfg: GPUGKEJob.Config = self.config
+
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
+            volume_mounts = [
+                {
+                    "name": "nvidia-install-dir-host",
+                    "mountPath": "/usr/local/nvidia/lib64",
+                },
+                {
+                    "name": "tcpx-socket",
+                    "mountPath": "/run/tcpx",
+                },
+            ]
+
+            command = [
+                "bash",
+                "-c",
+                """
+                /tcpgpudmarxd/build/app/tcpgpudmarxd --gpu_nic_preset a3vm \
+                    --gpu_shmem_type fd --uds_path /run/tcpx" \
+                    --setup_param "--verbose 128 2 0" &
+                while [ ! -f /run/tcpx/terminated ]; do sleep 10; done; "
+                """,
+            ]
+
+            return dict(
+                name="tcpx-daemon",
+                image="us-docker.pkg.dev/gce-ai-infra/gpudirect-tcpx/tcpgpudmarxd-dev:v2.0.11",
+                securityContext={"privileged": True},
+                command=command,
+                env=[{"name": "LD_LIBRARY_PATH", "value": "/usr/local/nvidia/lib64"}],
+                volumeMounts=volume_mounts,
+            )
+
+        return {}
+
+    def _build_main_container(self) -> Nested[Any]:
+        """Builds the config for the container running the job.
+
+        Returns:
+            A nested dict corresponding to a k8s Container config.
+        """
+        cfg: GPUGKEJob.Config = self.config
+        volume_mounts = []
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
+            volume_mounts.extend(
+                [
+                    {"name": "tcpx-socket", "mountPath": "/run/tcpx"},
+                    {"name": "shared-memory", "mountPath": "/dev/shm"},
+                    {"name": "nvidia-install-dir-host", "mountPath": "/usr/local/nvidia/lib64"},
+                    {"name": "tcpx-nccl-plugin-volume", "mountPath": "/usr/local/tcpx"},
+                ]
+            )
+
+        env_vars: Dict[str, str] = {}
+        env_vars["DISTRIBUTED_COORDINATOR"] = f"{cfg.name}-job-0-0.{cfg.name}:8080"
+        env_vars["NUM_PROCESSES"] = f"{cfg.accelerator.num_replicas}"
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
+            env_vars["LD_LIBRARY_PATH"] = "/usr/local/tcpx/lib64:/usr/local/nvidia/lib64"
```

Comment:
nit -- `env_vars.update({ ... })` may be slightly easier to read.

Reply:
fixed